documentation/optee_design.md

```diff
@@ -611,68 +431,68 @@ struct optee_header {
+what follows the header and the following `init_size` bytes to
+`(init_load_addr_hi << 32 | init_load_addr_lo)`. `init_mem_usage` is used by the
+loader to be able to check that there's enough physical memory available for
+OP-TEE to be able to initialize at all. The loader supplies the address of the
+first byte following what wasn't copied in r0/x0 and jumps to the load address
+to start OP-TEE.
 
 ## Initializing the pager
-The pager is initialized as early as possible during boot in order to
-minimize the "init" area. As a consequence of this the old initialization
-has been broken up a bit since the pager initialization depends on some
-internal APIs being available.
-
-The global variable `tee_mm_vcore` describes the virtual memory range that
-is covered by the level 2 translation table supplied to `tee_pager_init()`.
+The pager is initialized as early as possible during boot in order to minimize
+the "init" area. As a consequence of this the old initialization has been broken
+up a bit since the pager initialization depends on some internal APIs being
+available. The global variable `tee_mm_vcore` describes the virtual memory range
+that is covered by the level 2 translation table supplied to `tee_pager_init()`.
 
 ### Assign pageable areas
-A virtual memory range to be handled by the pager is registered with a
-call to `tee_pager_add_area()`.
-```
+A virtual memory range to be handled by the pager is registered with a call to
+`tee_pager_add_area()`.
+
+```c
 bool tee_pager_add_area(tee_mm_entry_t *mm, uint32_t flags, const void *store,
-                    const void *hashes);
+			const void *hashes);
 ```
-which takes a pointer to `tee_mm_entry_t` to tell
-the range, flags to tell how memory should be mapped (readonly, execute
-etc), and pointers to backing store and hashes of the pages.
+
+which takes a pointer to `tee_mm_entry_t` to tell the range, flags to tell how
+memory should be mapped (readonly, execute etc), and pointers to backing store
+and hashes of the pages.
 
 ### Assign physical pages
 Physical SRAM pages are supplied by calling `tee_pager_add_pages()`
-```
+
+```c
 void tee_pager_add_pages(tee_vaddr_t vaddr, size_t npages, bool unmap);
 ```
-`tee_pager_add_pages()` expects all physical memory to be mapped in the
-level 2 translation table already. `tee_pager_add_pages()` takes physical
-address stored in the entry mapping the virtual address "vaddr" and
-"npages" entries after that and uses it to map new pages when needed. The
-unmap parameter tells whether the pages should be unmapped immediately
-since they doesn't contain initialized data or be kept mapped until they
-need to be recycled.
 
-The pages in the "init" area are supplied with unmap == false since those
-page have valid content and are in use.
+`tee_pager_add_pages()` expects all physical memory to be mapped in the level 2
+translation table already. `tee_pager_add_pages()` takes physical address stored
```

Comment:
Remove "`tee_pager_add_pages()` expects all physical memory to be mapped in the level 2 translation table already".